src/proxy/processors/push-action/parsePush.ts

```diff
@@ -36,33 +30,25 @@ async function exec(req: any, action: Action): Promise<Action> {
   const step = new Step('parsePackFile');
   try {
     if (!req.body || req.body.length === 0) {
-      step.log('No data received in request body.');
-      step.setError('Your push has been blocked. No data received in request body.');
-      action.addStep(step);
-      return action;
+      throw new Error('No body found in request');
     }
     const [packetLines, packDataOffset] = parsePacketLines(req.body);
     const refUpdates = packetLines.filter((line) => line.includes(BRANCH_PREFIX));
 
     if (refUpdates.length !== 1) {
       step.log('Invalid number of branch updates.');
       step.log(`Expected 1, but got ${refUpdates.length}`);
-      step.setError(
+      throw new Error(
```

Comment:
Is there a reason (other than simplifying the code somewhat) to use `throw new Error` instead of `step.setError` here and in other spots in `parsePush.ts`?